rero_ils/modules/items/serializers/csv.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+# Copyright (C) 2020 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Item serializers."""
+
+import csv
+
+from invenio_i18n.ext import current_i18n
+from invenio_records_rest.serializers.csv import CSVSerializer, Line
+
+from rero_ils.filter import format_date_filter
+from rero_ils.modules.documents.api import search_document_by_pid
+from rero_ils.modules.documents.utils import title_format_text_head
+from rero_ils.modules.loans.api import search_loans_for_item
+from rero_ils.modules.locations.api import LocationsSearch
+
+
+class ItemCSVSerializer(CSVSerializer):
+    """Serialize and filter item circulation status."""
+
+    def transform_search_hit(
+        self, pid, record_hit, links_factory=None, **kwargs
+    ):
+        """Transform search result hit into an intermediate representation."""
+        hit = self.preprocess_search_hit(
+            pid, record_hit, links_factory=links_factory, **kwargs
+        )
+        return hit
+
+    def preprocess_search_hit(self, pid, record_hit, links_factory=None,
+                              **kwargs):
+        """Prepare a record hit from Elasticsearch for serialization."""
+        record = record_hit['_source']
+        item_pid = pid.pid_value
+
+        # process location
+        locations_map = kwargs.get('locations_map')
+        record['location_name'] = locations_map[
+            record.get('location').get('pid')]
+
+        # retrieve and process document
+        document = search_document_by_pid(record['document']['pid'])
+        record['document_title'] = title_format_text_head(document.title,
```

Comment:
why not `ui_title_text` as the JSON serializer?

Reply:
Because `document_title` used as key in the csv header and we get data values with csv header keys.